android/app/src/main/java/org/fossasia/openevent/adapters/viewholders/SocialLinkViewHolder.java

```diff
@@ -0,0 +1,113 @@
+import org.fossasia.openevent.utils.ConstantStrings;
+import org.fossasia.openevent.utils.Utils;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class SocialLinkViewHolder extends RecyclerView.ViewHolder {
+
+    @BindView(R.id.img_social_link)
+    protected ImageView imageView;
+
+    @BindView(R.id.layout_social_link_parent)
+    protected FrameLayout layout;
+
+    private SocialLink socialLink;
+    private Context context;
+
+    public SocialLinkViewHolder(View itemView, Context context) {
+        super(itemView);
+        ButterKnife.bind(this, itemView);
+
+        this.context = context;
+        imageView.setOnClickListener(view -> {
+            if (socialLink != null && !Utils.isEmpty(socialLink.getLink())) {
+                setUpCustomTab(socialLink.getLink());
+            }
+        });
+    }
+
+    private void setUpCustomTab(String url) {
+
+        Uri uri = Uri.parse(url);
+
+        CustomTabsIntent.Builder intentBuilder = new CustomTabsIntent.Builder();
+        intentBuilder.setToolbarColor(ContextCompat.getColor(context, R.color.color_primary));
+        intentBuilder.setSecondaryToolbarColor(ContextCompat.getColor(context, R.color.color_primary_dark));
+        intentBuilder.setStartAnimations(context, R.anim.slide_in_right, R.anim.slide_out_left);
+        intentBuilder.setExitAnimations(context, android.R.anim.slide_in_left, android.R.anim.slide_out_right);
+
+        CustomTabsIntent customTabsIntent = intentBuilder.build();
+        customTabsIntent.launchUrl(context, uri);
+    }
+
+    public void bindSocialLink(SocialLink socialLink) {
+        this.socialLink = socialLink;
+        setImageDrawable();
+    }
+
+    private void setImageDrawable() {
+        String name = this.socialLink.getName();
+        String link = this.socialLink.getLink();
+
+        if (name == null || link == null) {
+            showView(false);
+            return;
+        }
+
+        showView(true);
+
+        switch (name) {
+            case ConstantStrings.SOCIAL_LINK_GITHUB:
+                imageView.setImageDrawable(getDrawable(R.drawable.ic_github_24dp));
+                break;
+            case ConstantStrings.SOCIAL_LINK_TWITTER:
+                imageView.setImageDrawable(getDrawable(R.drawable.ic_twitter_24dp));
+                break;
+            case ConstantStrings.SOCIAL_LINK_FACEBOOK:
+                imageView.setImageDrawable(getDrawable(R.drawable.ic_facebook_24dp));
+                break;
+            case ConstantStrings.SOCIAL_LINK_LINKEDIN:
+                imageView.setImageDrawable(getDrawable(R.drawable.ic_linkedin_24dp));
+                break;
+            case ConstantStrings.SOCIAL_LINK_YOUTUBE:
+                imageView.setImageDrawable(getDrawable(R.drawable.ic_youtube_24dp));
+                break;
+            case ConstantStrings.SOCIAL_LINK_GOOGLE:
+                imageView.setImageDrawable(getDrawable(R.drawable.ic_google_plus_24dp));
+                break;
```

Comment:
This code is duplicated in other classes, right?

Rather make a util method which returns a drawable ID when passed a name